src/PotentialCalculation/PotentialCalculationSetup/PotentialCalculationSetupCartesian3D.jl

```diff
@@ -290,14 +291,27 @@ function PotentialCalculationSetup(det::SolidStateDetector{T}, grid::CartesianGr
                 paint_contacts = Val(paint_contacts)  )
         rbpotential = RBExtBy2Array( potential, grid )
         rbpoint_types = RBExtBy2Array( point_types, grid )
+        
+        new_imp_scale::Array{T, 4} = if ismissing(imp_scale) || is_weighting_potential 
+            broadcast(is_pn_junction_point_type, rbpoint_types)
+        else
+            new_imp_scale = RBExtBy2Array(imp_scale, grid)
+            for i in eachindex(new_imp_scale)
+                if !is_pn_junction_point_type(rbpoint_types[i]) 
+                    new_imp_scale[i] = 0
+                end
+            end

```

Comment:
Instead of the loop:
```julia
new_imp_scale .*= broadcast(is_pn_junction_point_type, rbpoint_types)
```
??

Also, I guess julia is smart enough that we don't need to define the type of 0, i.e. `T(0)` or `zero(T)`?